### DISCUSSION TITLE
GPT : Challenge me in Solidity

### DISCUSSION THREAD

**Discussion by ironcladmerc** [Cyfrin/foundry-full-course-cu#1109] [General]
I've been playing around with the new GPT's in ChatGPT4, I created a GPT called "Challenge Me in Solidity", you can ask it for an easy, medium, hard or security challenge. It will then lay out the coding challenge for what it wants you to do, and then you can go off to Remix or your IDE to complete the challenge. Once you're done, you can copy/paste your code into the GPT and it'll give you feedback on how you did.  I thought it would be good for practice.

Here's the link:
https://chat.openai.com/g/g-ToTJ0LIFD-challenge-me-in-solidity

Hope you like it!

**Comment by 0xusmanf** [comment]
Great initiative, would be very beneficial for those looking to challenge themselves. Unfortunately, I don't have access to custom GPTs so I can't say if it already has the following features which I am going to suggest, if I may.

1. Apart from `easy`, `medium`,` hard` or `security` also make it topic centric, for example, if someone wants to practice mappings, it should suggest challenges focused on mappings. Similarly ERC20, ERC721, Proxies, Oracles etc.
2. For security if someone wants to practice a reentrancy attack it should suggest challenges that include reentrancy vulnerability so you can try and solve it.

**Reply by ironcladmerc** [reply]
You can ask it for a certain type of challenge apart from just the difficulty. like you can say "give me an easy ERC20 challenge" and it will do it.  I can't update it anymore without creating a new one, apparently they don't let you update your own GPTs once they've been finalized or once you leave the build screen for it
